packages/x-components/src/components/location-injector.vue

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+  import { Prop, Provide, Component } from 'vue-property-decorator';
+  import { NoElement } from '../components/no-element';
+  import { FeatureLocation } from '../types';
+  import BaseEventButton from './base-event-button.vue';
+
+  /**
+   * Location Injector component.
+   * This component injects the location with value passed as prop.
+   *
+   * @public
+   */
+  @Component({
+    components: { BaseEventButton }
+  })
+  export default class LocationInjector extends NoElement {
+    /**
+     * The {@link FeatureLocation} to inject.
+     *
+     * @public
+     */
+    @Prop({ required: true })
+    @Provide('location')
```

Comment:
`Vue`'s `Provide` uses by default the same key as the variable name ([docs](https://github.com/kaorun343/vue-property-decorator#Provide)). So, in this case, you don't need to specify it again.

```suggestion
    @Provide()
```